JavaScript/Node.js/processcsv.py

```diff
@@ -60,22 +60,32 @@ def processcsvfile(fname,seperator,outdir,outfname):
     container = args.container
     pattern = args.pattern   
 
+    container_client = block_blob_service.get_container_client(container)
+
     print("Processing files from container : " + str(container))
     if pattern is not None:
-        blob_list = block_blob_service.list_blobs(container_name=container, prefix=pattern)
+        blob_list = container_client.list_blobs(name_starts_with=pattern)
     else:
-        blob_list = block_blob_service.list_blobs(container_name=container)
+        blob_list = container_client.list_blobs()
     
     for blob in blob_list:
         print("Processing blob : " + blob.name)
-        blob_name = getfilename(blob.name)
+        blob_name = getfilename(blob.name).split('.')[0]
         downloadedblob = "downloaded_" + blob_name 
-        block_blob_service.get_blob_to_path(container_name=container,blob_name=blob.name, file_path=downloadedblob, open_mode='w')
+
+         #Download the blob locally
+        blob_client = container_client.get_blob_client(blob.name)
+        with open(downloadedblob, "wb") as my_blob:
+            downloaded_blob_stream = blob_client.download_blob()
+            my_blob.write(downloaded_blob_stream.readall())
+
         if pattern is not None:
             json_outpath = processcsvfile(fname=downloadedblob,seperator="|",outfname=blob_name,outdir='jsonfiles/' + container + "/" + pattern)
-            print("uploading blob" + json_outpath)
-            block_blob_service.create_blob_from_path(container_name=container,blob_name=str(pattern+ 'json/' + blob_name + ".json"),file_path=json_outpath) 
+            print("uploading blob " + json_outpath)
+            with open(json_outpath, "rb") as data:
+                container_client.upload_blob(name=str(pattern + 'json/' + blob_name + ".json"), data=data)
         else:
             json_outpath = processcsvfile(fname=downloadedblob,seperator="|",outfname=blob_name,outdir='jsonfiles/' + container + "/")
-            print("uploading blob" + json_outpath)
-            block_blob_service.create_blob_from_path(container_name=container,blob_name=str('json/' + blob_name + ".json"),file_path=json_outpath)            
\ No newline at end of file
+            print("uploading blob " + json_outpath)
+            with open(json_outpath, "rb") as data:
+                container_client.upload_blob(name=str('json/' + blob_name + ".json"), data=data)
```

Comment:
Seems like most of this is being repeated in both conditions.

Reply:
Good catch! Will consolidate that logic.